File: FruitRacers.Backend/FruitRacers.Backend.Core/Entities/Product.cs
using System;
using System.Collections.Generic;

namespace FruitRacers.Backend.Core.Entities
{
    public class Product
    {
        public Product()
        {
            OrderDetails = new HashSet<OrderDetail>();
            Prices = new HashSet<Price>();
        }

        public int ProductId { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool? IsEnabled { get; set; }
        public bool? IsLegal { get; set; }
        public bool IsDeleted { get; set; }
        public int SupplierId { get; set; }
        public int CategoryId { get; set; }
        public int? ImageId { get; set; }

        public virtual Supplier Supplier { get; set; }
        public virtual Category Category { get; set; }
        public virtual ICollection<OrderDetail> OrderDetails { get; set; }
        public virtual ICollection<Price> Prices { get; set; }
        public virtual Image Image { get; set; }
    }
}

File: FruitRacers.Frontend/components/product.js
class Product {

    constructor(
        id = 0,
        name = "Product name",
        description = "",
        categories = [],
        manufacturer = null,
        price = 0,
        unit = "ND",
        minQuantity = 0,
        image = ""
    ) {
        this.id = id;
        this.name = name;
        this.description = description;
        this.categories = categories;
        this.manufacturer = manufacturer;
        this.price = price;
        this.unit = unit;
        this.minQuantity = minQuantity;
        this.image = image;
    }

    get id() { return this._id; }
    set id(value) { this._id = value; }

    get name() { return this._name; }
    set name(value) { this._name = value; }

    get description() { return this._description; }
    set description(value) { this._description = value; }

    get categories() { return this._categories; }
    set categories(value) { this._categories = value; }

    get manufacturer() { return this._manufacturer; }
    set manufacturer(value) { this._manufacturer = value; }

    get price() { return this._price; }
    set price(value) { this._price = value; }

    get unit() { return this._unit; }
    set unit(value) { this._unit = value; }

    get minQuantity() { return this._minQuantity; }
    set minQuantity(value) { this._minQuantity = value; }

    get image() { return this._image; }
    set image(value) { this._image = value; }
}